lib/iopromise/data_loader.rb

```diff
@@ -3,13 +3,20 @@
 module IOPromise
   module DataLoader
     module ClassMethods
-      def attr_promised_data(*args)
+      def attr_async_data(attr_name, build_func = nil)
         @promised_data_keys ||= []
-        @promised_data_keys.concat(args)
+        @promised_data_keys << attr_name
```

Comment:
Should we rename all `@promised_*` variables to match the new method name?

Reply:
:+1: Agreed, let's ditch `promised` for `async` all over here to be consistent.